vault/request_handling.go

```diff
@@ -772,6 +777,17 @@ func (c *Core) handleRequest(ctx context.Context, req *logical.Request) (retResp
 		return nil, nil, ctErr
 	}
 
+	// Updating in-flight request data with client/entity ID
+	inFlightReqID, ok := ctx.Value("in-flight-reqID").(string)
+	if ok {
+		switch {
+		case req.ClientID != "":
+			c.UpdateInFlightReqData(inFlightReqID, req.ClientID)
+		case req.EntityID != "":
```

Comment:
I think if there's an entityid it'll be present in req.ClientID too, so you shouldn't need this switch.